### DISCUSSION TITLE
Modal component not work in Livewire

### DISCUSSION THREAD

**Discussion by tegarmonster** [htmlstreamofficial/preline#416] [Q&A]
Hi,

I was wondering why, Modal Component works for the first load if using Livewire,

I was adding this script on app.js file

`import 'preline';

document.addEventListener('livewire:navigate', () => {
    HSStaticMethods.autoInit();
});

document.addEventListener('livewire:navigating', () => {
    HSStaticMethods.autoInit();
});

document.addEventListener('livewire:navigated', () => {
    HSStaticMethods.autoInit();
});
`

also include this script into main blade component
`<head>
    
        @vite(['resources/css/app.css', 'resources/js/app.js'])
</head>`

but the Javascript cannot load if I move to another page or navigate.. any ideas?

**Comment by elasawilmar7** [comment]
Hi, Did you manage to solve the problem?

**Reply by tegarmonster** [reply]
nope, but I was try to create custom component to handle that
